### PR TITLE
#4452 neo modal

### LINKED ISSUE #4452: Refactor: Oruga-ui modal wrapper in histoire
- Create `NeoModal` component based on https://oruga.io/components/Modal.html
- design might be in figma
- replace `b-modal` in `GalleryItemSharBtn` with this new component

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #4452
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: 
[Payout](https://beta.kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screen Shot 2023-04-07 at 5 38 05 PM

### CODE REVIEW

**Comment by prachi00** [comment]
not sure about the overlay color in the background, it is too dark right now, @exezbcz can you tell me what it should be for modals?

File: libs/ui/src/components/NeoModal/NeoModal.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+<script lang="ts" setup>
+import { OModal } from '@oruga-ui/oruga'
+
+const props = defineProps<{
+  value: boolean
+}>()
+
+const emit = defineEmits(['close'])
+
+const isImageModalActive = useVModel(props, 'value')
```
Comment (Jarsen136) [inline libs/ui/src/components/NeoModal/NeoModal.vue:19]:
```suggestion
const isModalActive = useVModel(props, 'active')
```

File: libs/ui/src/components/NeoModal/NeoModal.vue [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    @close="updateClose">
+    <slot />
+  </o-modal>
+</template>
+
+<script lang="ts" setup>
+import { OModal } from '@oruga-ui/oruga'
+
+const props = defineProps<{
+  value: boolean
```
Comment (Jarsen136) [inline libs/ui/src/components/NeoModal/NeoModal.vue:14]: Is it better to use name `visible/active`? wdyt

**Review by Jarsen136 — CHANGES_REQUESTED** [review]
There still some buefy modal usage in our project. Could we provide a oruga version?
image



ref: [ oruga modal programmatically section ](https://oruga.io/components/modal.html)
```
      oruga.modal.open({
        component: ModalForm,
        trapFocus: true
      })
```

**Comment by exezbcz** [comment]
@prachi00 same as for sidebar

File: libs/ui/src/components/NeoModal/NeoModal.vue (hunk 1 above)
Comment (vikiival) [reply libs/ui/src/components/NeoModal/NeoModal.vue:19]: agree with constant name

File: libs/ui/src/components/NeoModal/NeoModal.vue (hunk 2 above)
Comment (vikiival) [reply libs/ui/src/components/NeoModal/NeoModal.vue:14]: The way that @prachi00 used that could be implemented with v-model

**Comment by vikiival** [comment]
Does it work with dark mode?

File: libs/ui/src/components/NeoModal/NeoModal.vue (hunk 2 above)
Comment (Jarsen136) [reply libs/ui/src/components/NeoModal/NeoModal.vue:14]: Yes, I mean we could rename `value` to `active` for improved clarity.

**Comment by prachi00** [comment]
> 
Screen Shot 2023-04-08 at 4 30 25 PM
yes

**Comment by prachi00** [comment]
> 

can we do this version in a separate issue, as it is part of another module called @oruga-ui/oruga-next and I was thinking of creating a hook called useModal for this but we would have to convert the components to vue 3 to use it so it could drag a bit longer

**Comment by Jarsen136** [comment]
> > 
> 
> can we do this version in a separate issue, as it is part of another module called @oruga-ui/oruga-next and I was thinking of creating a hook called useModal for this but we would have to convert the components to vue 3 to use it so it could drag a bit longer

Yes, let's leave it another issue.

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Comment by vikiival** [comment]
👁️ But it wasn't defined in the scope

Screenshot 2023-04-10 at 15 23 23

**Comment by vikiival** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 32.74 USD/KSM ~ 0.916 $KSM
🧗 [EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU](https://kusama.subscan.io/account/EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)
🔗 [0xca65ece68ff4157d75d902715483c2c525e1ac6b6dc4e59ab3f56348881da35e](https://kusama.subscan.io/extrinsic/0xca65ece68ff4157d75d902715483c2c525e1ac6b6dc4e59ab3f56348881da35e)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**